pyvrp/crossover/tests/test_selective_route_exchange.py

```diff
@@ -0,0 +1,173 @@
+from numpy.testing import assert_equal
+
+from pyvrp import Individual, PenaltyManager, XorShift128
+from pyvrp.crossover import selective_route_exchange as srex
+from pyvrp.tests.helpers import read
+
+
+def test_srex_move_all_routes():
+    """
+    Tests if SREX produces an offspring that is identical to the second parent
+    when all routes are replaced during crossover.
+    """
+    data = read("data/OkSmall.txt")
+    pm = PenaltyManager(data.vehicle_capacity)
+    rng = XorShift128(seed=4)  # nMovedRoutes = 3
+
+    indiv1 = Individual(data, pm, [[1], [2], [3, 4]])
+    indiv2 = Individual(data, pm, [[1, 2], [3], [4]])
+    parents = (indiv1, indiv2)
+
+    offspring = srex(parents, data, pm, rng)
+
+    assert_equal(offspring.get_routes(), indiv2.get_routes())
```

Comment:
Question: how does this work? I get that they are all replaced, but why?

Reply:
Oh because `nMovedRoutes = 3`?

Reply:
If so, can we maybe have something along the lines of: "this RNG seed will cause `nMovedRoutes = 3`", in this test and in others? It wasn't immediately obvious to me what hte comment behind the rng meant.

Reply:
I don't like that we tie these lower-level parameters to the RNG state so tightly. Perhaps we should split off SREX's implementation into a separate function that takes `startA` and `nMovedRoutes` as arguments, with a shim around it that keeps the original interface.

But that's maybe something for a new issue.